sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -505,7 +505,7 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
     val schema = StructType(Seq(
       StructField("f", StructType(Seq(
         StructField("a", StringType, nullable = true),
-        StructField("b", IntegerType, nullable = false)
+        StructField("b", IntegerType, nullable = true)
```

Comment:
With the new null check, we will trigger error earlier than this test expected. This test is testing the `AssertNotNull` expression, which is used for converting nullable column to not-nullable object field(like primitive int).